### ISSUE TITLE
No job functions found (for fresh clone)

### ISSUE THREAD

**Issue by alexjamesbrown** [JamesRandall/FunctionMonkey#64]
Getting `No job functions found ` on running FunctionMonkey.Tests.Integration.Functions
(fresh download of master)

Had to change target runtime to netcoreapp2.2 (think this was due to a recent change in the azure func runtime) 

Azure Functions Core Tools (2.6.1080 Commit hash: 509d59b7eb6fbfb89f70ce96ad22e00f8f7bafdd)
Function Runtime Version: 2.0.12427.0

> [10/05/2019 12:25:06] No job functions found. Try making your job classes and methods public. If you're using binding extensions (e.g. Azure Storage, ServiceBus, Timers, etc.) make sure you've called the registration method for the extension(s) in your startup code (e.g. builder.AddAzureStorage(), builder.AddServiceBus(), builder.AddTimers(), etc.).
> [10/05/2019 12:25:06] Host initialized (94ms)
> [10/05/2019 12:25:06] Host started (102ms)
> [10/05/2019 12:25:06] Job host started
> Hosting environment: Production
> Content root path: C:\Users\alex\Desktop\FunctionMonkey\Tests\FunctionMonkey.Tests.Integration.Functions\bin\Debug\netcoreapp2.2

**Comment by JamesRandall** [comment]
Ahhh yes. They are a bit weird. They don't reference the Compiler NuGet package as I usually use them to test the version of the compiler I am working on in the solution.

There's an added complexity in that the build process will clean out files when you Debug the functions (and as the package isn't referenced they won't get dropped back in).

Building this stuff as part of a dev flow is a bit weird as a result.

The way I do it is to set things up like this:

1. Remove the FunctionMonkey.Tests.Integration.Functions from the build configuration and set it as the default startup project.

2. Set the debug arguments for FunctionMonkey.Compiler to point to the output of the above project (when I get to my other PC I'll drop in what mine looks like).

After doing this I take the following approach to make a change.

1. Build the solution.

2. Right click -> build the FunctionMonkey.Tests.Integration.Functions project

3. Right click -> debug the FunctionMonkey.Compiler project

4. Hit Run

I need to get round to automating this a bit better.

**Comment by mcantillon** [comment]
Hey,

I got this in a new project when I forgot to reference "FunctionMonkey.Compiler". Not sure this is the issue however it may help a path of investigation. :)

I noticed when I tried to run the tests that the compiler failed to build because I didn't have Yarn installed. Once installed I could build it and see it pulling in the swagger module.
If you do a solution clean and rebuild all, is it successful?

Matt
